packages/editor/src/store/effects/reusable-blocks.js

```diff
@@ -61,8 +62,13 @@ export const fetchReusableBlocks = async ( action, store ) => {
 				} ),
 			];
 		} else {
+			const perPage = Platform.select( {
+				// Unbounded queries don't work on native so we can only fetch the maximum items for now
+				native: 100,
+				web: -1,
+			} );
```

Comment:
As far as I checked if `per_page=-1` query parameter is included in the request, one of the `api-fetch` middleware items [(`fetch-all-middleware`)](https://github.com/WordPress/gutenberg/blob/master/packages/api-fetch/src/middlewares/fetch-all-middleware.js) consider it as an unbounded query.  This means that it will try to fetch all the available items of this resource by potentially doing multiple requests with the maximum items per page until it reaches the end.

As stated in the code:
```
// The REST API enforces an upper limit on the per_page option. To handle large
// collections, apiFetch consumers can pass `per_page=-1`; this middleware will
// then recursively assemble a full response array from all available pages.
```

The problem is that this middleware requires the full response object to apply its logic, for example it needs to read the `link` header to know the next page. Unfortunately the mobile version of `api-fetch` doesn't return the full response, it only returns the response's body so we can't rely on this logic, which means that we can't use that query parameter because it raises an exception.

My first approach was to find a workaround in the Gutenberg project but for the mobile version the request logic is handled in the native side, maybe in the future that part should be reviewed because this could be a limitation if we need to do unbounded queries.

For now, as a workaround, what I've done is fetch the maximum items per page that should be enough for Reusable blocks.

Reply:
I also wrote a [P2 post](https://mobilegutenberg.wordpress.com/2020/09/21/unbounded-queries-support/) about this just in case we want to tackle it in the future.